code/modules/chemistry/Reagents-Drugs.dm

```diff
@@ -490,6 +490,84 @@ datum
 				..()
 				return
 
+		drug/solipsizine
+			name = "solipsizine"
+			id = "solipsizine"
+			description = "A highly potent hallucinogenic substance that causes intense delirium and acute inability to percieve others."
+			reagent_state = LIQUID
+			depletion_rate = 0.1
+			addiction_prob = 8
+			fluid_r = 200
+			fluid_g = 120
+			fluid_b = 120
+			transparency = 50
+			var/counter = 1
+			var/list/invisible_people = list()
+			var/image/override_img = null
+			var/list/image/override_list = list()
+
+			on_mob_life(var/mob/M, var/mult = 1)
+				if(!M) M = holder.my_atom
+				src.counter += 1 * mult //around half realtime
+
+				if(probmult(3))
+					boutput(M, pick("<span class='notice'>You feel eerily alone..</span>",\
+									"<span class='notice'>You feel like everything's gone silent.</span>",\
+									"<span class='notice'>Everything seems so quiet all of a sudden.</span>",\
+									"<span class='notice'>You can hear your heart beating.</span>",\
+									"<span class='notice'>Something is wrong.</span>"))
+				else if(probmult(3))
+					M.emote(pick("shiver","shudder","drool"))
+
+				if(counter > 15) //turn everyone into nothing
+					if(M.ear_damage < 15)
+						M.take_ear_damage(3 * mult, 1) //makes it so you can't hear people after a bit
+
+					var/list/candidates = list() //adds people just out of sight to the list of "make go away"
+					candidates = by_type[/mob/living/carbon/human].Copy() - invisible_people - viewers(M)
```

Comment:
```suggestion
					candidates = by_type[/mob/living/carbon/human] - invisible_people - viewers(M)
```
should be no need to copy here as you aren't modifying it, and copying it will be expensive